src/js/utils.js

```diff
@@ -106,7 +120,32 @@ this.exports = {
     return false;
   },
   
+  hasHeatMode: function(thermostat) {
+    return thermostat.settings.heatStages > 0 || thermostat.settings.hasHeatPump;
+  },
+  
+  hasCoolMode: function(thermostat) {
+    return thermostat.settings.coolStages > 0 || thermostat.settings.hasHeatPump;
+  },
+  
+  hasAuxHeatMode: function(thermostat) {
```

Comment:
I extracted these functions from the ecobee angular app so they should be correct.